src/shared_realm.cpp

```diff
@@ -518,6 +538,13 @@ VersionID Realm::read_transaction_version() const
     return static_cast<Transaction&>(*m_group).get_version_of_current_transaction();
 }
 
+uint_fast64_t Realm::get_number_of_versions() const
+{
+    verify_open();
+    check_read_write(this);
```

Comment:
check_read_write() is now a very misnamed function and it makes all the code using it confusing. check_write() and check_read_write() should be renamed to reflect what they're actually checking (check_write() might be correct but I'm not sure, check_read_write() is actually checking if m_group is a Transaction).